engine/class_modules/paladin/sc_paladin_holy.cpp

```diff
@@ -450,16 +455,90 @@ struct lights_hammer_t : public paladin_spell_t
 
 // Light of Dawn (Holy) =======================================================
 
-struct light_of_dawn_t : public paladin_heal_t
+struct light_of_dawn_t : public holy_power_consumer_t<paladin_heal_t>
 {
   light_of_dawn_t( paladin_t* p, const std::string& options_str )
-    : paladin_heal_t( "light_of_dawn", p, p->find_class_spell( "Light of Dawn" ) )
+    : holy_power_consumer_t( "light_of_dawn", p, p->find_class_spell( "Light of Dawn" ) )
   {
     parse_options( options_str );
 
     aoe = 6;
+  }
+
+  void execute() override
+  {
+    holy_power_consumer_t::execute();
+
+    // deal with awakening
+    if ( p()->talents.awakening->ok() )
+    {
+      if ( rng().roll( p()->talents.awakening->effectN( 1 ).percent() ) )
+      {
+        buff_t* main_buff           = p()->buffs.avenging_wrath;
+        timespan_t trigger_duration = timespan_t::from_seconds( p()->talents.awakening->effectN( 2 ).base_value() );
+        if ( main_buff->check() )
+        {
+          p()->buffs.avengers_might->extend_duration( p(), trigger_duration );
+        }
+        else
+        {
+          main_buff->trigger( 1, buff_t::DEFAULT_VALUE(), -1.0, trigger_duration );
+        }
+      }
+    }
+  }
+};
+
+// Word Of Glory (Holy) =======================================================
+
+struct word_of_glory_t : public holy_power_consumer_t<paladin_heal_t>
```

Comment:
There's already a word_of_glory_t defined in sc_paladin_protection.cpp, it would be better to merge the two and place the resulting struct in sc_paladin.cpp

Reply:
Additionally, you could move the awakening proc to holy_power_consumer_t and gate it behind a bool that defaults to false and is set to true in WoG and LoD constructors. Or keep it the current way, it's your choice and I don't think one is better than the other

Reply:
I think the first option is probably the most logical so i'll do that

Reply:
Just to note. Before someone comes to yell at me about `p()->specialization() == PALADIN_HOLY` is not needed
Im going to leave it in there since prot paladin had it in there by default and it should keep some form of readability + the ability to blindly look at it and say `oh this is something for holy. let me not nuke it completely`